### PR TITLE
[LT] Turn expand into a view op

### PR DESCRIPTION
**Pull request by alanwaketan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75350
* #75344
* #75247
* #75246
* __->__ #75245
* #75230

Summary:
Somehow op expand was implemented as a regular op instead of a view op.
This commit corrects this mistake such that our alias check infra can
correctly detected expanded tensor as an alias of the pre-expanded one.

Test Plan:
./build/bin/test_lazy --gtest_filter=LazyOpsTest.TestExpand*

Differential Revision: [D35388566](https://our.internmc.facebook.com/intern/diff/D35388566)

### CODE REVIEW

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35388566).

File: torch/csrc/lazy/core/lazy_view.cpp [hunk 1]
```diff
@@ -126,6 +133,12 @@ Value ApplyUpdate(Value ir_value, const Alias::UpdateData& update_data) {
             view_info.diagonal->dim1,
             view_info.diagonal->dim2);
         break;
+      case ViewInfo::Type::kExpand:
```
Comment (wconstab) [inline torch/csrc/lazy/core/lazy_view.cpp:136]: it's not obvious to me if this is OK.

if someone first calls expand, then modifies the result, the 'view' system would track that modification and make sure any other users also get it.  in your way, directly returning an 'Expand' node here, does it lose the view tracking? or is this somehow interposing late enough that the view tracking already happened?

File: torch/csrc/lazy/core/view_ops/expand.h [hunk 2]
```diff
@@ -7,6 +7,9 @@
 namespace torch {
 namespace lazy {
 
+// Expand can be constructed as a regular view op, or a scalar expand.
```
Comment (wconstab) [inline torch/csrc/lazy/core/view_ops/expand.h:10]: expand isn't really a 'regular view op' in the current PR is it? or maybe i'm mistaken

File: torch/csrc/lazy/core/lazy_view.cpp (hunk 1 above)
Comment (alanwaketan) [reply torch/csrc/lazy/core/lazy_view.cpp:136]: It won't work. @Krovatkin is handling this op now.

File: torch/csrc/lazy/core/view_ops/expand.h (hunk 2 above)
Comment (alanwaketan) [reply torch/csrc/lazy/core/view_ops/expand.h:10]: I mean those expand ops that are created through `GetIrValueForExpandedScalar`.

**Comment by alanwaketan** [comment]
Close this, and please refer to #75333 for follow up.

**Pull request closed without merging**